src/basics.md

```diff
@@ -277,6 +278,54 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of allowed characters
+
+[![rand-badge]][rand]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`] or [`choose`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+# 
+# error_chain! {
+#     foreign_links {
+#         String(std::string::FromUtf8Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let s: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", s);
+
+    let alphanum_bytestring =
+        String::from_utf8((b'a'..b'z')
+        .chain(b'A'..b'Z')
+        .chain(b'0'..b'9')
+        //.chain([b'!', b'?', b'%'])
+        .collect())?;
+
+    let mut rand_bytevec: Vec<u8> = Vec::new();
+
+    for _ in 0..30 {
```

Comment:
If the iterator approach would be used, pretty much all of this code would be not needed.